libminifi/include/utils/OsUtils.h

```diff
@@ -29,8 +29,17 @@ namespace OsUtils {
 /// Resolves a user ID to a username
 extern std::string userIdToUsername(const std::string &uid);
 
-/// Returns memory usage in bytes, including shared memory
-uint64_t getMemoryUsage();
+/// Returns physical memory usage by the current process in bytes
+uint64_t getCurrentProcessPhysicalMemoryUsage();
```

Comment:
I also looked into getting information about virtual memory but its meaning is wildly inconsistent between platforms and usually doesn't really hold useful information.